File: LC 880.cpp
class Solution {
public:
    string decodeAtIndex(string s, int k) {
        string ans;
        long long n = s.length(), m = 0;
        int j = -1;
        for (int i = 0; i < n; ++i) {
            if (isalpha(s[i])) {
                m++;
                if (m == k) {
                    cout << s[i];
                    ans = s[i];
                    break;
                }
            } else {
                int t = s[i] - '0';
                m *= t;
                if (m >= k) {
                    j = i;
                    break;
                }
            }
        }
        for (; j >= 0; j--) {
            k %= m;
            if (k == 0 && isalpha(s[j]))
                return (string)"" + s[j];
            if (isdigit(s[j])) {
                m /= s[j] - '0';
            } else {
                m--;
            }
        }
        return ans;
    }
};

File: 1000/LC 880.go
func decodeAtIndex(s string, k int) string {
    j, m := -1, 0
    var ans string
    for i, c := range s {
        if c >= 'a' && c <= 'z' {
            m++
            ans = string(c)
        } else {
            x := int(c - '0')
            m *= x
            if m > k {
                j = i
                break
            }
        }
        if m == k {
            return ans
        }
    }

    for ; j >= 0; j-- {
        k %= m
        if k == 0 && s[j] >= 'a' && s[j] <= 'z' {
            return string(s[j])
        } else if s[j] >= '2' && s[j] <= '9' {
            m /= int(s[j] - '0')
        } else {
            m--
        }
    }

    return ""
}